goroutine/gid_test.go

```diff
@@ -35,8 +40,14 @@ func TestID(t *testing.T) {
 	wg.Wait()
 }
 
+var testID int64
+
 // goid returns the goroutine ID extracted from a stack trace.
 func goid() int64 {
+	if IsTinyGo {
+		// pretty horrible test that aligns with the implementation, but at least it tests we get 1,2,3... different numbers.
+		return atomic.AddInt64(&testID, 1)
```

Comment:
This is the old way to use atomic

Now you could/should use `var testID atomic.Int64`

Then here you can use


```suggestion
		return testID.Add(1)
```

More information here
https://pkg.go.dev/sync/atomic#Int64

Reply:
I'm not going to pretend I knew and did it on purpose, yet

https://github.com/fortio/log/blob/main/go.mod#L3

vs

<img width="154" alt="Screenshot 2024-07-17 at 12 03 49 PM" src="https://github.com/user-attachments/assets/526e8caa-78a8-4151-9648-9a61f47305de">

Reply:
Haha